### DISCUSSION TITLE
Search user by metadata key

### DISCUSSION THREAD

**Discussion by ranjiteds** [zitadel/zitadel#6066] [Q&A]
Hi, 

We are lookin into a business case where were have multiple child entities (login) that can represent a user. Its like a user can be logged in using different login entities, and it should be logged in as one parent user. 

Ex - 
User A ( Main Zitadal user )
   - Entity 1 ( is authenticated from external server  )
   - Entity 2 
  
So if a user uses User A, or Entity 1 or 2, logged in session is for the user A

This could be possible if I can mange to use Metadata of User A and assign (key/value ) pair to entity 1 or 2 login id and tokens. 

Use cases -

- User logs in with User A ( This is ok )
    User logs in with normal authetication and we get all the metadata for this user. and can show it on client side.

- User try to log in with Entity 1 
  1. In this case what we can do is to search the parent user A by metadata key ( login id for entity 1 ) and get the parent user, creates its session. but I could not find a way to search the user by metadata key. ( which is suppose to be unique accross all the users).
  2.  Also in order to search the metadata or get any details of a user, i will need a user that can access all the user related data at global level, is this possible? Can I create a user that can access all the user in a perticular project?  

Or if someone can help me to find a better approach? 

Thanks a lot

**Comment by hifabienne** [comment]
Hei @ranjiteds 
You said that entity 1 and entity 2 are authenticated in an external system. This sounds like external identity provider to me.
You can connect different external providers to your user. For example a google and a github account.
So your User A is the identity in ZITADEL.
Entity 1 and Entity 2 are entities in an external system that are linked to User A. 
Would that be a possibility?
https://zitadel.com/docs/guides/integrate/identity-providers

We do not yet have the possibility to search users by metadata

**Reply by ranjiteds** [reply]
thanks i am going to try this

**Reply by ranjiteds** [reply]
If I can link my user to external authentication mechanism then I wont need metadata search

**Comment by ranjiteds** [comment]
I have tested this case, and it works perfect for our use case. I tried it with Google IDP and I am able to make it work. 

I am testing JWT provider now, following 
https://zitadel.com/docs/concepts/structure/jwt_idp

Steps I have taken

Configured JWT IDP -
![image](https://github.com/zitadel/zitadel/assets/94850276/2041e038-ef5f-4600-b11b-49ac79073219)

JWT endpoint sends an OK status with JWT inside the authorization header with name - "x-auth-token".

So if I am right, according to the documentation 
https://zitadel.com/docs/concepts/structure/jwt_idp,  it says -

"ZITADEL will do so and redirect to the preconfigured JWT Endpoint. While the endpoint is behind the WAF, ZITADEL is able to receive a JWT from a defined http header. It will then validate its signature, which might require to call the configured Keys Endpoint. If the signature is valid and token is not expired, ZITADEL will then use the token and the enclosed sub claim as if it was an id_token returned from an OIDC IDP: It will try to match a user's external identity and if not possible, create a new user with the information of the provided token. Prerequisite for this is that the IDP setting autoregister is set to true."

So when i try to log in user it shows me "JWT" in other users section on login page and when I click on it, the page then redirects me to the JWT endppoint that I have configured. "/api/jwt".  

My questions - 
1. Zitadal login flow should validate the JWT from the endpoint and should get the user in "sub" attribute of jwt payload, checks if the user exists in the Zitadle system and redirects you to user link page. but this is not happening. 
2. Am I missing something here with JWT structure? Or I probably got it all wrong with JWT IDP flow. 

Thanks

**Reply by hifabienne** [reply]
@livio-a Can you help here?
